### PR TITLE
test: Fix wait_for_debug_log UnicodeDecodeError

### LINKED ISSUE #24575: feature_init intermittent issue:                                    UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 1215-1217: unexpected end of data
Presumably because a Unicode character was only written partially.

```
 test  2022-03-15T19:13:39.817000Z TestFramework (ERROR): Unexpected exception caught during testing 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main
                                       self.run_test()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/feature_init.py", line 79, in run_test
                                       node.start(extra_args=['-txindex=1', '-blockfilterindex=1', '-coinstatsindex=1'])
                                     File "/usr/lib/python3.8/contextlib.py", line 120, in __exit__
                                       next(self.gen)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_node.py", line 442, in wait_for_debug_log
                                       log = dl.read()
                                     File "/usr/lib/python3.8/codecs.py", line 322, in decode
                                       (result, consumed) = self._buffer_decode(data, self.errors, final)
                                   UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 1215-1217: unexpected end of data
```

https://cirrus-ci.com/task/5773197005029376?logs=ci#L5215

### PR DESCRIPTION
**Pull request by maflcko**
Fix the intermittent `UnicodeDecodeError` when the debug log is truncated on an (multi-byte) unicode character by treating everything as bytes.

Also, remove the `ignore_case` option and the`re.search+re.escape` wrap. All of this is unused and doesn't exist on raw byte strings.

Fixes https://github.com/bitcoin/bitcoin/issues/24575

### CODE REVIEW

**Comment by Adminsupportteam** [comment]
Kindly follow the step on this link to talk to the customer care service directly on your issues anyone asking you to DM is a scam 

https://direct.lc.chat/14201763/

**Comment by Admindesk24** [comment]
Kindly follow the step on this link to talk to the customer service directly on your issues 

https://direct.lc.chat/14201793/

**Review by jonatack — COMMENTED** [review]
ACK

File: test/functional/feature_init.py [hunk 1]
```diff
@@ -49,33 +49,33 @@ def check_clean_start():
...
+            b'msghand thread start',
+            b'net thread start',
+            b'addcon thread start',
         ]
         if self.is_wallet_compiled():
-            lines_to_terminate_after.append('Verifying wallet')
+            lines_to_terminate_after.append(b'Verifying wallet')
 
         for terminate_line in lines_to_terminate_after:
             self.log.info(f"Starting node and will exit after line '{terminate_line}'")
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply test/functional/feature_init.py]: thx, fixed

**Comment by jonatack** [comment]
ACK fa74b63c01db412f6a4378cb669d89496a89d02e

**Review by brunoerg — APPROVED** [review]
ACK fa74b63c01db412f6a4378cb669d89496a89d02e

**Comment by PiRK** [comment]
@MarcoFalke I think line 451 of test_node.py is now broken. `log` is now no longer a `string`, but `bytes`.

```
            if time.time() >= time_end:
                print_log = " - " + "\n - ".join(log.splitlines())
                break
```

`mypy` detected this when I tried adding a typehint (`List[bytes]`) to the `expected_msgs`  parameter of `wait_for_debug_log`,.

**Pull request merged**